lib/core/rename.h

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include <vector>
+
+#include "scipp-core_export.h"
+#include "scipp/core/except.h"
+#include "scipp/units/dim.h"
+
+namespace scipp::core {
+
+namespace detail {
+template <class T>
+[[nodiscard]] T rename_dims(const T &obj,
+                            const std::vector<std::pair<Dim, Dim>> &names,
```

Comment:
Use `std::span` instead? (And in functions that use it)

Reply:
Doesn't that have trouble when passing an initializer_list?

Reply:
Yes. Is it used in many places? I only remember seeing it in `groupby`. Not sure if it is worth avoiding the creation of a vector elsewhere. The main use case seems to be in Python where it has to create a vector anyway.

Reply:
Mostly in unit tests.